bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -171,30 +174,41 @@ private void pollingCode() {
         updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER_L3,
                 new QuantityType<>(payload.getActivePowerL3W(), Units.WATT));
 
-        updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
-                new QuantityType<>(payload.getTotalGasM3(), SIUnits.CUBIC_METRE));
-
-        // 210119164000
+        // If no data from the gas meter is present, the json value will be null, which means gson ignores it,
+        // leaving the value in the payload object at 0.
         long dtv = payload.getGasTimestamp();
-        long seconds = dtv % 100;
-
-        dtv /= 100;
-        long minutes = dtv % 100;
-
-        dtv /= 100;
-        long hours = dtv % 100;
-
-        dtv /= 100;
-        long day = dtv % 100;
-
-        dtv /= 100;
-        long month = dtv % 100;
-
-        dtv /= 100;
-        long year = dtv + 2000; // Where (When?) have I seen this before?
-
-        DateTimeType dtt = DateTimeType
-                .valueOf(String.format("%04d-%02d-%02dT%02d:%02d:%02d", year, month, day, hours, minutes, seconds));
-        updateState(HomeWizardBindingConstants.CHANNEL_GAS_TIMESTAMP, dtt);
+        if (dtv > 0) {
+            updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
+                    new QuantityType<>(payload.getTotalGasM3(), SIUnits.CUBIC_METRE));
+
+            // 210119164000
+            long seconds = dtv % 100;
+
+            dtv /= 100;
+            long minutes = dtv % 100;
+
+            dtv /= 100;
+            long hours = dtv % 100;
+
+            dtv /= 100;
+            long day = dtv % 100;
+
+            dtv /= 100;
+            long month = dtv % 100;
+
+            dtv /= 100;
+            long year = dtv + 2000;
+
+            String dateString = String.format("%04d-%02d-%02dT%02d:%02d:%02d", year, month, day, hours, minutes,
+                    seconds);
+            try {
+                DateTimeType dtt = DateTimeType.valueOf(dateString);
+                updateState(HomeWizardBindingConstants.CHANNEL_GAS_TIMESTAMP, dtt);
+                updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
+                        new QuantityType<>(payload.getTotalGasM3(), SIUnits.CUBIC_METRE));
+            } catch (java.time.format.DateTimeParseException e) {
```

Comment:
Thanks for the fix! I have one small suggestion, can you import the class instead?

```suggestion
            } catch (DateTimeParseException e) {
```

It would be even better if you'd create a `DateTimeType` using a `ZonedDateTime`, then you don't need to format the values and parse them again, which will save some CPU cycles. :slightly_smiling_face:

Reply:
Thanks! I've applied the suggestions.